grails-app/services/de/iteratec/osm/measurement/environment/QueueAndJobStatusService.groovy

```diff
@@ -72,7 +72,7 @@ class QueueAndJobStatusService {
         if (locationsResponse != null) {
             locationsResponse.data.location.each { locationTagInXml ->
                 List<String> browserNames = locationTagInXml.Browsers.size() != 0 ? locationTagInXml.Browsers.toString().split(",") : [locationTagInXml.Browser.toString()]
-                List<Browser> browsersForLocation = browserService.findAllByNameOrAlias(browserNames)
+                List<Browser> browsersForLocation = browserService.findAllByName(browserNames)
                 browsersForLocation.each { Browser currentBrowser ->
                     String uniqueIdentfierForServer = locationTagInXml.id.toString().endsWith(":${currentBrowser.name}") ?: locationTagInXml.id.toString() + ":${currentBrowser.name}"
```

Comment:
Here the same deprecated logic is used as it was used in `LocationPersisterService.groovy` and may lead top the same inconsistencies.
You can simply generate the `uniqueIdentfierForServer` from XML data and find locations by `uniqueIdentfierForServer` to have a safe match.

Reply:
Wouldn't that also mean that the method findAllByNameOrAlias isn't even needed anymore? If the locations are queried with data directly from the XML the list of browsers is not needed for anything else and the method only returns the active locations. So the method could be removed?